packages/discovery-provider/ddl/functions/chat_allowed.sql

```diff
@@ -26,6 +26,19 @@ BEGIN
     RETURN TRUE;
   END IF;
 
+  -- existing chat takes priority over permissions
+  SELECT COUNT(*) > 0 INTO can_message
```

Comment:
do we need to take `cleared_history_at` into account for these? ie. existence of messages after `cleared_history_at` -> `can_message = true`, otherwise false?

maybe too complicated.. also maybe something we need to ask design about

Reply:
I think it's irrelevant because if I add this it will always evaluate to true.
You'll have to block to prevent future messages.
That might actually also be true for your earlier PR too... I didn't think about that at the time 🤔 

```
AND (member_b.cleared_history_at is null or member_b.cleared_history_at < now())
```

Reply:
hmm.. was thinking 
```
  JOIN chat_message USING (chat_id)
  WHERE member_a.user_id = from_user_id
    AND member_b.user_id = to_user_id
    AND (member_b.cleared_history_at is null or chat_message.created_at > member.cleared_history_at)
  ;
does that make sense?

Reply:
thinking is:
- user A and user B have some chat history
- user A clears chat history
- user A closes their inbox perms
in this scenario, should user B be able to message user A?
if yes, then we're good.
if no, then we should add the `cleared_history_at` checks i mentioned, such that now:
- user B can't message user A, until
- user A messages user B again
- user A and B can now chat as normal